app/src/main/java/msi/paria/trainingapp/MainActivity.kt

```diff
@@ -1,11 +1,86 @@
 package msi.paria.trainingapp
 
 import android.os.Bundle
+import android.util.Log
 import androidx.activity.ComponentActivity
+import androidx.compose.ui.Modifier
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+
 
 class MainActivity : ComponentActivity() {
+
+    private lateinit var viewModelImpl: MainViewModelImpl
+
+    private val _state = mutableStateOf(MainPageState())
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        setContent {
+            Surface(modifier = Modifier.fillMaxSize(), color = Color.White) {
+                view()
+            }
+        }
+
+        viewModelImpl = object : MainViewModelImpl {
```

Comment:
We must directly use the view model here. There is no need to use Impl

Reply:
we don't have any ViewModel in this state, and we must use abstraction. for test real time functionality of view, we used simple implementation.